File: discourse/cache.js
const redis = require('redis');

const client = redis.createClient();
client.connect().then(() => {
    console.log('Redis connected');
}).catch((err) => {
    console.error(err);
})

const namespace = 'discourse-node:';

function SetCache(key, value) {
    return client.set(namespace + key, JSON.stringify(value));
}

function GetCache(key) {
    return client.get(namespace + key);
}

module.exports = {
    SetCache,
    GetCache,
};

File: discourse/cache.go
package discourse

import (
	"context"

	"github.com/redis/go-redis/v9"
)

var ctx = context.Background()

var redisClient = redis.NewClient(&redis.Options{
	Addr:     "localhost:6379",
	Password: "", // no password set
	DB:       0,  // use default DB
})

var namespace = "discourse-go:"

func SetCache(key string, value string) {
    redisClient.Set(ctx, namespace+key, value, 0)
}

func GetCache(key string) (string, error) {
    return redisClient.Get(ctx, namespace+key).Result()
}
